themes/default/content/docs/intro/concepts/stack.md

```diff
@@ -700,6 +700,275 @@ as your current stack (i.e. when deploying the "staging" stack of the above prog
+    return fmt.Sprintf("logs/%s.log", ip)
+}).(StringOutput)
+logFile := s3.NewBucketObject(ctx, "log", &s3.BucketObjectArgs{
+    // ...
+    Key: logKey,
+})
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var infra = new StackReference(...);
+var ip = infra.GetOutput("privateIp");
+var logKey = ip.Apply(ip => $"logs/{ip}.log");
+var logFile = new Aws.S3.BucketObject("log", new Aws.S3.BucketObjectArgs
+{
+    // ...
+    Key = logKey,
+});
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+StackReference infra = new StackReference(...);
+Output<String> ip = infra.getOutput("privateIp");
+Output<String> logKey = ip.apply(ip -> String.format("logs/%s.log", ip));
+BucketObject logFile = new BucketObject("log", new BucketObjectArgs.Builder()
+    .key(logKey)
+    .build());
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+On the other hand, as an example of using **`getOutputDetails`**,
+suppose that your referenced stack creates a VPC
+and exports a list of public subnets as a JSON-serialized string,
+and you want to add a bastion host to each subnet.
+With `getOutputDetails`, this would look something like this:
+
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+const infra = new pulumi.StackReference(...);
+const subnetsJSON = await infra.getOutputDetails("subnets");
+const subnets = JSON.parse(subnetsJSON.value);
+for (let i = 0; i < subnets.length; i++) {
+    const subnet = subnets[i];
+    const host = new aws.ec2.Instance(`bastion-${i}`, {
+        // ...
+        subnetId: subnet.id,
+    });
+    // ...
+}
+```
+
+Note that your Pulumi program must export a top-level `async` function
+to be able to use the `await` operator.
+
+```javascript
+export = async () => {
+    // ...
+}
+```
+
+See [Javascript Entrypoint](https://www.pulumi.com/docs/intro/languages/javascript/#entrypoint)
+for more information.
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const infra: StackReference = new pulumi.StackReference(...);
+const subnetsJSON: StackReferenceOutputDetails = await infra.getOutputDetails("subnets");
+const subnets = JSON.parse(subnetsJSON.value);
+for (let i = 0; i < subnets.length; i++) {
+    const subnet = subnets[i];
+    const host = new aws.ec2.Instance(`bastion-${i}`, {
+        // ...
+        subnetId: subnet.id,
+    });
+    // ...
+}
+```
+
+Note that your Pulumi program must export a top-level `async` function
+to be able to use the `await` operator.
+
+```javascript
+export = async () => {
+    // ...
+}
+```
+
+See [Javascript Entrypoint](https://www.pulumi.com/docs/intro/languages/javascript/#entrypoint)
+for more information.
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+{{% notes "info" %}}
+This functionality is not currently supported in Python.
+Please vote on [pulumi/pulumi#12172](https://github.com/pulumi/pulumi/issues/12172)
```

Comment:
```suggestion
Progress is tracked on [pulumi/pulumi#12172](https://github.com/pulumi/pulumi/issues/12172)
```

Reply:
given that this isn't lack of desire nor trivial, i think tracking is more appropriate than voting ...